tests/Aspire.Hosting.Tests/Python/AddPythonProjectTests.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Tests.Helpers;
+using Xunit;
+using Microsoft.Extensions.DependencyInjection;
+using System.Text.RegularExpressions;
+
+namespace Aspire.Hosting.Tests.Python;
+
+public class AddPythonProjectTests
+{
+    private static readonly Regex s_pythonExecutablePattern  = new Regex("python(\\.(exe|bat|cmd|sh|ps1))?$", RegexOptions.IgnoreCase);
+    private static readonly Regex s_telemetryExecutablePattern = new Regex("opentelemetry-instrument(\\.(exe|bat|cmd|sh|ps1))?$", RegexOptions.IgnoreCase);
+    private static readonly string s_playgroundDirectory = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../../../../playground/python/"));
+
+    [LocalOnlyFact("python")]
+    public void AddPythonProject_SetsResourcePropertiesCorrectly()
+    {
+        var pythonProjectDirectory = Path.Combine(s_playgroundDirectory, "script_only");
+        var appBuilder = DistributedApplication.CreateBuilder();
+
+        appBuilder.AddPythonProject("python", pythonProjectDirectory, "main.py");
+
+        var app = appBuilder.Build();
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+        var executableResources = appModel.GetExecutableResources();
+
+        var pythonProjectResource = Assert.Single(executableResources);
+
+        Assert.Equal("python", pythonProjectResource.Name);
+        Assert.Equal(pythonProjectDirectory, pythonProjectResource.WorkingDirectory);
+    }
```

Comment:
I would love to see some pure unit tests like this https://github.com/dotnet/aspire/blob/main/tests/Aspire.Hosting.Tests/Garnet/AddGarnetTests.cs

Reply:
I need a little advice on this. The python virtual env layout varies across platforms, on Windows there's a `scripts` directory while on linux and mac there's a `bin` directory for the executables. I could add OS detection and assume that the scripts/binaries are in the right spot. But I felt that it would be nicer to actually check on disk if the binaries are there.

Because of this choice I figured it would make sense to have local facts with an actual python environment instead of writing pure unit-tests.

I can make purer unit-tests by either: Making the assumption about the virtual env layout and not checking the disk. Or I can write temp dummy files to disk to make the unit-test pass. 

What do you prefer? Or do you have other ideas?

Reply:
What I'm doing with the incoming WithDockerfile(...) extension is creating a temp directory and dropping the minimum number of files I need in there. If the tests are going to actually spin up Python that is what you'd need to do.